### PR TITLE
Fix #631, Fix #884

### LINKED ISSUE #631: inserting a simple entry into warnings filter, the new entry is not work
```
E:\2.5test\ironpythonCompatableFor2.5\25\Lib>ipy  
IronPython console: 2.0  
Copyright (c) Microsoft Corporation. All rights reserved.  

>>> import warnings  
warnings.filters  
[('ignore', None, , None, 0), ('ignore', None, , None, 0)]  
oldfilters = warnings.filters  
warnings.simplefilter('error',ImportWarning)  
__import__("site-packages")  
Traceback (most recent call last):  
File , line 0, in ##281  
ImportError: No module named site-packages  
warnings.filters = oldfilters  
^Z  
```

```
E:\2.5test\ironpythonCompatableFor2.5\25\Lib>python  
Python 2.5 (r25:51908, Sep 19 2006, 09:52:17) [MSC v.1310 32 bit (Intel)] on win32  
Type "help", "copyright", "credits" or "license" for more information.  
import warnings  
warnings.filters  
[('ignore', None, , None, 0), ('ignore', None, , None, 0)]  
oldfilters = warnings.filters  
warnings.simplefilter('error',ImportWarning)  
__import__("site-packages")  
Traceback (most recent call last):  
File "", line 1, in  
File "C:\Python25\lib\warnings.py", line 62, in warn  
globals)  
File "C:\Python25\lib\warnings.py", line 102, in warn_explicit  
raise message  
ImportWarning: Not importing directory 'site-packages': missing ** init ** .py  
warnings.filters=oldfilters  
```
## Work Item Details

**Original CodePlex Issue:**    [Issue 23911](http://ironpython.codeplex.com/workitem/23911)
**Status:** Active
**Reason Closed:**  Unassigned
**Assigned to:**    Unassigned
**Reported on:**    Jul 28, 2009 at 12:08 AM
**Reported by:**    dfugate
**Updated on:** Feb 22, 2013 at 2:12 AM
**Updated by:** jdhardy
**Custom value:**   Reported internally at Microsoft.
**Test:**   test_import.py
**CreatedDate:**    1/8/2008
**NewInternalID:**  409731
**OldInternalID:**  365388
**AreaPath:**   IronPython\Python25

### LINKED ISSUE #884: Regressions in test_import when using 2.7 standard lib
C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib >ipy -X:Frames
test\test_import.py  
test_case_sensitivity (test.test_import.ImportTests) ... ok  
test_double_const (test.test_import.ImportTests) ... ok  
test_execute_bit_not_copied (test.test_import.ImportTests) ... skipped 'test
meaningful only on posix systems'  
test_failing_import_sticks (test.test_import.ImportTests) ... ok  
test_failing_reload (test.test_import.ImportTests) ... ok  
test_imp_module (test.test_import.ImportTests) ... ok  
test_import (test.test_import.ImportTests) ... ERROR  
test_import_by_filename (test.test_import.ImportTests) ... ok  
test_import_initless_directory_warning (test.test_import.ImportTests) ...
skipped ' [ http://ironpython.codeplex.com/workitem/28171
](http://ironpython.codeplex.com/workitem/28171)  
'  
test_import_name_binding (test.test_import.ImportTests) ... ok  
test_infinite_reload (test.test_import.ImportTests) ... ok  
test_module_with_large_stack (test.test_import.ImportTests) ... ok  
test_basics (test.test_import.PycRewritingTests) ... FAIL  
test_foreign_code (test.test_import.PycRewritingTests) ... ERROR  
test_incorrect_code_name (test.test_import.PycRewritingTests) ... ERROR  
test_module_without_source (test.test_import.PycRewritingTests) ... ERROR  
test_trailing_slash (test.test_import.PathsTests) ... ok  
test_absolute_import_without_future (test.test_import.RelativeImportTests) ...
skipped ' [ http://ironpython.codeplex.com/workitem/
](http://ironpython.codeplex.com/workitem/)  
28171'  
test_issue3221 (test.test_import.RelativeImportTests) ... ok  
test_relimport_star (test.test_import.RelativeImportTests) ... ok  
# 
## ERROR: test_import (test.test_import.ImportTests)

Traceback (most recent call last):  
File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\test\tes
t_import.py", line 86, in test_import  

```
test_with_extension(os.extsep + "py")
```

File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\test\tes
t_import.py", line 76, in test_with_extension  

```
imp.reload(mod)
```

SystemError: module source file not found  
# 
## ERROR: test_foreign_code (test.test_import.PycRewritingTests)

Traceback (most recent call last):  
File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\test\tes
t_import.py", line 337, in test_foreign_code  

```
py_compile.compile(self.file_name)
```

File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\py_compi
le.py", line 126, in compile  

```
marshal.dump(codeobject, fc)
```

ValueError: unmarshallable object  
# 
## ERROR: test_incorrect_code_name (test.test_import.PycRewritingTests)

Traceback (most recent call last):  
File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\test\tes
t_import.py", line 320, in test_incorrect_code_n  
ame  

```
py_compile.compile(self.file_name, dfile="another_module.py")
```

File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\py_compi
le.py", line 126, in compile  

```
marshal.dump(codeobject, fc)
```

ValueError: unmarshallable object  
# 
## ERROR: test_module_without_source (test.test_import.PycRewritingTests)

Traceback (most recent call last):  
File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\test\tes
t_import.py", line 328, in test_module_without_s  
ource  

```
py_compile.compile(self.file_name, dfile=target)
```

File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\py_compi
le.py", line 126, in compile  

```
marshal.dump(codeobject, fc)
```

ValueError: unmarshallable object  
# 
## FAIL: test_basics (test.test_import.PycRewritingTests)

Traceback (most recent call last):  
File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\test\tes
t_import.py", line 314, in test_basics  

```
self.assertEqual(mod.module_filename, self.compiled_name)
```

AssertionError:
'C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\ [ C:\\vsl0\
\dlr\\External.LCA_RESTRICTED\\Languages\\IronPython\\27\\Lib\\@test_2 
](mailto:C:\\vsl0\\dlr\\External.LCA_RESTRICTED\\Languages\\IronPyth
on\\27\\Lib\\@test_2) [truncated]... != 'C:\vsl0\dl  
r\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\ [
r\\External.LCA_RESTRICTED\\Languages\\IronPython\\27\\Lib\\@test_2 ](ma
ilto:r\\External.LCA_RESTRICTED\\Languages\\IronPython\\27\\Lib\\@
test_2) [truncated]...  
- [ C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\@test_2260_tmp\unlikely_module_name.py+ ](mailto:C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\@test_2260_tmp\unlikely_module_name.py+) C:\vsl0\dlr\External.L  
  [ CA_RESTRICTED\Languages\IronPython\27\Lib\@test_2260_tmp\unlikely_module_nam
  e.pyc ](mailto:CA_RESTRICTED\Languages\IronPython\27\Lib\@test_2260_tmp\
  unlikely_module_name.pyc) ?
  
  ```
                                                   +
  ```
## 

Ran 20 tests in 1.269s  

FAILED (failures=1, errors=4, skipped=3)  
Traceback (most recent call last):  
File "test\test_import.py", line 466, in <module>  
File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\test\tes
t_import.py", line 457, in test_main  
File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\test\tes
t_support.py", line 1078, in run_unittest  
File "C:\vsl0\dlr\External.LCA_RESTRICTED\Languages\IronPython\27\Lib\test\tes
t_support.py", line 1061, in _run_suite  
test.test_support.TestFailed: multiple errors occurred  
## Work Item Details

**Original CodePlex Issue:**    [Issue 28596](http://ironpython.codeplex.com/workitem/28596)
**Status:** Proposed
**Reason Closed:**  Unassigned
**Assigned to:**    Unassigned
**Reported on:**    Aug 17, 2010 at 6:30 AM
**Reported by:**    darb
**Updated on:** Feb 22, 2013 at 2:10 AM
**Updated by:** jdhardy

### PR DESCRIPTION
**Pull request by slide**
test_import now passes (with skips for pyc/pyo files)
test_warnings is much closer

### CODE REVIEW

File: Test/IronPython.tests [hunk 1]
```diff
@@ -3072,7 +3072,7 @@
           <MaxDuration>1200000</MaxDuration>
           <LongRunning>false</LongRunning>
           <Disabled>true</Disabled> <!-- FAILED -->
-          <RequiresAdmin>false</RequiresAdmin>
+          <RequiresAdmin>true</RequiresAdmin>
```
Comment (slozier) [inline Test/IronPython.tests]: Is this really what you wanted to change? Or did you want to reenable the test?

File: Languages/IronPython/IronPython/Hosting/PythonOptionsParser.cs [hunk 2]
```diff
@@ -201,6 +201,15 @@ public sealed class PythonOptionsParser : OptionsParser<PythonConsoleOptions> {
                     break;
 
                 default:
+                    if(arg.StartsWith("-W")) {
```
Comment (slozier) [inline Languages/IronPython/IronPython/Hosting/PythonOptionsParser.cs:204]: Not specific to this issue, but should we be adding a special handler for the `-m` argument as well?

File: Test/IronPython.tests (hunk 1 above)
Comment (slide) [reply Test/IronPython.tests]: For now, yes. I need to test a few more things.

File: Languages/IronPython/IronPython/Hosting/PythonOptionsParser.cs (hunk 2 above)
Comment (slide) [reply Languages/IronPython/IronPython/Hosting/PythonOptionsParser.cs:204]: That's probably a good idea. Do you want to file an issue for it, or should I?

**Pull request merged**